src/main/java/com/mraof/minestuck/data/MSBlockStateProvider.java

```diff
@@ -0,0 +1,1215 @@
+		
+		slabWithItem(MSBlocks.BLACK_CHESS_BRICK_SLAB, MSBlocks.BLACK_CHESS_BRICKS);
+		slabWithItem(MSBlocks.DARK_GRAY_CHESS_BRICK_SLAB, MSBlocks.DARK_GRAY_CHESS_BRICKS);
+		slabWithItem(MSBlocks.LIGHT_GRAY_CHESS_BRICK_SLAB, MSBlocks.LIGHT_GRAY_CHESS_BRICKS);
+		slabWithItem(MSBlocks.WHITE_CHESS_BRICK_SLAB, MSBlocks.WHITE_CHESS_BRICKS);
+		slabWithItem(MSBlocks.FLOWERY_MOSSY_STONE_BRICK_SLAB, MSBlocks.FLOWERY_MOSSY_STONE_BRICKS.getId().withSuffix("1"));
+		slabWithItem(MSBlocks.COARSE_STONE_SLAB, MSBlocks.COARSE_STONE);
+		slabWithItem(MSBlocks.COARSE_STONE_BRICK_SLAB, MSBlocks.COARSE_STONE_BRICKS);
+		slabWithItem(MSBlocks.SHADE_SLAB, MSBlocks.SHADE_STONE);
+		slabWithItem(MSBlocks.SHADE_BRICK_SLAB, MSBlocks.SHADE_BRICKS);
+		slabWithItem(MSBlocks.FROST_TILE_SLAB, MSBlocks.FROST_TILE);
+		slabWithItem(MSBlocks.FROST_BRICK_SLAB, MSBlocks.FROST_BRICKS);
+		slabWithItem(MSBlocks.BLACK_STONE_SLAB, MSBlocks.BLACK_STONE);
+		slabWithItem(MSBlocks.BLACK_STONE_BRICK_SLAB, MSBlocks.BLACK_STONE_BRICKS);
+		slabWithItem(MSBlocks.MYCELIUM_SLAB, MSBlocks.MYCELIUM_STONE);
+		slabWithItem(MSBlocks.MYCELIUM_BRICK_SLAB, MSBlocks.MYCELIUM_BRICKS);
+		slabWithItem(MSBlocks.CHALK_SLAB, MSBlocks.CHALK);
+		slabWithItem(MSBlocks.CHALK_BRICK_SLAB, MSBlocks.CHALK_BRICKS);
+		slabWithItem(MSBlocks.PINK_STONE_SLAB, MSBlocks.PINK_STONE);
+		slabWithItem(MSBlocks.PINK_STONE_BRICK_SLAB, MSBlocks.PINK_STONE_BRICKS);
+		slabWithItem(MSBlocks.BROWN_STONE_SLAB, MSBlocks.BROWN_STONE);
+		slabWithItem(MSBlocks.BROWN_STONE_BRICK_SLAB, MSBlocks.BROWN_STONE_BRICKS);
+		slabWithItem(MSBlocks.GREEN_STONE_SLAB, MSBlocks.GREEN_STONE);
+		slabWithItem(MSBlocks.GREEN_STONE_BRICK_SLAB, MSBlocks.GREEN_STONE_BRICKS.getId(),
+				texture("horizontal_green_stone_bricks"), texture("polished_green_stone"));
+		slabWithItem(MSBlocks.RAINBOW_PLANKS_SLAB, MSBlocks.RAINBOW_PLANKS);
+		slabWithItem(MSBlocks.END_PLANKS_SLAB, MSBlocks.END_PLANKS);
+		slabWithItem(MSBlocks.DEAD_PLANKS_SLAB, MSBlocks.DEAD_PLANKS);
+		slabWithItem(MSBlocks.TREATED_PLANKS_SLAB, MSBlocks.TREATED_PLANKS);
+		
+		directional(MSBlocks.TRAJECTORY_BLOCK, state -> {
+			Direction direction = state.getValue(TrajectoryBlock.FACING);
+			boolean powered = state.getValue(TrajectoryBlock.POWERED);
+			ResourceLocation modelId = MSBlocks.TRAJECTORY_BLOCK.getId();
+			modelId = direction.getAxis() == Direction.Axis.Y ? modelId.withSuffix("_vertical") : modelId.withSuffix("_horizontal");
+			ResourceLocation topTexture = texture(modelId.withSuffix("_top"));
+			modelId = powered ? modelId.withSuffix("_powered") : modelId.withSuffix("_unpowered");
+			topTexture = powered ? topTexture.withSuffix("_powered") : topTexture.withSuffix("_unpowered");
+			
+			if(direction.getAxis() == Direction.Axis.Y)
+				return models().cubeColumn(modelId.getPath(),
+						texture("redstone_machine_block"),
+						topTexture);
+			else
+				return models().getExistingFile(modelId);
+		}, BlockStateProperties.POWER);
```

Comment:
On first glance, this and the other machine/redstone blocks appears to be more convoluted than the manual method. Is there a certain advantage you feel this brings to more complex blocks like these?

Reply:
Yeah, I can see that some of these are problematic in their current state.
Once I'm at the end, I'll do a second sweep to keep, improve or get rid of these more complex model generation blocks. I can poke you to go through what's left once I've done so.